rust/gui-client/src-frontend/components/App.tsx

```diff
@@ -91,65 +104,59 @@ export default function App() {
       >
         <SidebarItems>
           <SidebarItemGroup>
-            <NavLink to="/overview">
-              {({ isActive }) => (
-                <SidebarItem active={isActive} icon={HomeIcon} as="div">
-                  Overview
-                </SidebarItem>
-              )}
-            </NavLink>
+            <SidebarItem
+              active={location.pathname.startsWith("/overview")}
+              icon={HomeIcon}
+              href="/overview"
+              onClick={handleClick}
+            >
+              Overview
+            </SidebarItem>
             <SidebarCollapse label="Settings" open={true} icon={Bars3Icon}>
-              <NavLink to="/general-settings">
-                {({ isActive }) => (
-                  <SidebarItem active={isActive} icon={CogIcon} as="div">
-                    General
-                  </SidebarItem>
-                )}
-              </NavLink>
-              <NavLink to="/advanced-settings">
-                {({ isActive }) => (
-                  <SidebarItem
-                    active={isActive}
-                    icon={WrenchScrewdriverIcon}
-                    as="div"
-                  >
-                    Advanced
-                  </SidebarItem>
-                )}
-              </NavLink>
+              <SidebarItem
+                active={location.pathname.startsWith("/general-settings")}
+                icon={CogIcon}
+                href="/general-settings"
+                onClick={handleClick}
+              >
+                General
+              </SidebarItem>
+              <SidebarItem
+                active={location.pathname.startsWith("/advanced-settings")}
+                icon={WrenchScrewdriverIcon}
+                href="/advanced-settings"
+                onClick={handleClick}
```

Comment:
It appears to me that a better abstraction would be if we had our own `ReactRouterSidebarItem` component that uses the two hooks internally and checks if its `href` is in the pathname. Then we'd only need to pass `icon` and `href` and children to each component.

Reply:
yeah good point.